rfd/0056-cockroach-backend.md

```diff
@@ -0,0 +1,70 @@
+---
+authors: Jim Bishopp (jim@goteleport.com)
+state: draft
+---
+
+# RFD 56 - CockroachDB Backend
+
+## What
+
+A CockroachDB backend extends Teleport with the ability to store data
+in a self-hosted CockroachDB cluster.
+
+## Why
+
+Supporting additional backends such as CockroachDB reduces onboarding time
+for Teleport customers by allowing them to use existing infrastructure.
+
+## Scope
+
+This RFD focuses on implementing a backend for a self-hosted CockroachDB cluster.
+
+Requirements:
+
+- Prioritize self-hosted CockroachDB.
+- Implementation should support CockroachDB node failures by automatically
```

Comment:
Can the driver handle this automatically or will we need to implement this ourselves?

Reply:
I'd like to hold off on this for now because I'm not sure of the value and am concerned about complexity. Connecting to multiple nodes assumes each node is consistent. This RFD focuses on postgres and cockroachdb, and supporting multiple nodes is really only a cockroachdb feature. We'd have to deal with scenarios like detecting whether the fallback node is in the same state as the original node. Considering most cloud providers have their own solutions for HA that allow clients to specify a single endpoint, this additional work would be of questionable value.